OpticalNavigation/core/find_algos/find_with_hough_transform_and_contours.py

```diff
@@ -0,0 +1,583 @@
+    areas = [circleArea(circle) for circle in circles[0]]
+    max_idx = np.argmax(areas)
+    circle = circles[0][max_idx]
+    xy = (int(circle[0]), int(circle[1]))
+    r = int(circle[2])
+    cv2.circle(img, xy, r, (255, 0, 0), 3)
+    cv2.imshow("name", img)
+    cv2.waitKey(0)
+    return xy, r
+
+
+def __findBody(img, thresh, body, w, h):
+    """
+    Finds the earth, sun, moon
+    """
+    percentWhite = cv2.countNonZero(thresh) / (thresh.shape[0] * thresh.shape[1])
+    if body == "e":
+        percentageThresh = EARTH_PERCENTAGE_THRESH
+        highThreshRed = cv2.inRange(img, (0, 0, 5), (255, 255, 255))
+        highThreshGreen = cv2.inRange(img, (0, 5, 0), (255, 255, 255))
+        highThreshBlue = cv2.inRange(img, (5, 0, 0), (255, 255, 255))
+        highThresh = highThreshRed + highThreshGreen + highThreshBlue
+    elif body == "s":
+        percentageThresh = SUN_PERCENTAGE_THRESH
+        highThresh = cv2.inRange(img, (225, 225, 225), (255, 255, 255))
+    else:
+        percentageThresh = MOON_PERCENTAGE_THRESH
+        highThresh = thresh
+    if percentWhite >= percentageThresh:
+        # TODO: Shift center based on aspect ratio, center of "mass"
+        # return __findMinEnclosingCircle(img, highThresh)
+        return __houghCircleWithContour(img, w, h, highThresh)
+
+
+# Threshold based primarily on blue and green channels
+# Percent of white pixels determines if earth detected
+def measureEarth(img, w, h):
+    EARTH_THRESH = (
+        (EARTH_B_LOW, EARTH_G_LOW, EARTH_R_LOW),
+        (EARTH_B_HIGH, EARTH_G_HIGH, EARTH_R_HIGH),
+    )
+    return __findBody(
+        img, cv2.inRange(img, EARTH_THRESH[0], EARTH_THRESH[1]), "e", w, h
+    )
+
+
+# Measure white pixels
+def measureSun(img, w, h):
+    SUN_THRESH = (
+        (SUN_B_LOW, SUN_G_LOW, SUN_R_LOW),
+        (SUN_B_HIGH, SUN_G_HIGH, SUN_R_HIGH),
+    )
+    return __findBody(img, cv2.inRange(img, SUN_THRESH[0], SUN_THRESH[1]), "s", w, h)
+
+
+# TODO add code for % white, but parameterize if we can to use it
+def measureMoon(img, w, h):
+    MOON_THRESH = (
+        (MOON_B_LOW, MOON_G_LOW, MOON_R_LOW),
+        (MOON_B_HIGH, MOON_G_HIGH, MOON_R_HIGH),
+    )
+    return __findBody(img, cv2.inRange(img, MOON_THRESH[0], MOON_THRESH[1]), "m", w, h)
+
+
+def find(src, camera_params: CameraParameters = CisLunarCameraParameters):
+    start_time = time.time()
+    cam = Camera(
+        radians(camera_params.hFov),
+        radians(camera_params.vFov),
+        camera_params.hPix,
+        camera_params.vPix,
+    )
+
+    # u is in body frame here
+    # Assumes only spinning about y-axis
+    u = np.array([0, 1, 0], dtype=np.float32)
+    camNum = int(re.search(r"[cam](\d+)", src).group(1))
+    if camNum == 1:
+        u = np.linalg.inv(camera_params.cam1Rotation).dot(u)
+    elif camNum == 2:
+        u = np.linalg.inv(camera_params.cam2Rotation).dot(u)
+    elif camNum == 3:
+        u = np.linalg.inv(camera_params.cam3Rotation).dot(u)
```

Comment:
This fixes the pyright error for me, replaces with a warning for "cam_u might be unbound", but I think that's acceptable
```suggestion
camNum = int(re.search(r"[cam](\d+)", src).group(1))
    assert camNum in [1, 2, 3], "Error in getting the camera number from filename"
    if camNum == 1:
        cam_DCM: np.ndarray = np.linalg.inv(camera_params.cam1Rotation)
        u_cam = cam_DCM.dot(u)
    elif camNum == 2:
        cam_DCM: np.ndarray = np.linalg.inv(camera_params.cam2Rotation)
        u_cam = cam_DCM.dot(u)
    elif camNum == 3:
        cam_DCM: np.ndarray = np.linalg.inv(camera_params.cam3Rotation)
        u_cam = cam_DCM.dot(u)
```

Reply:
Worked for me after changing np.array to np.ndarray in the class definition where u is